status/versions_test.go

```diff
@@ -0,0 +1,139 @@
+package status
+
+import (
+	"testing"
+)
+
+func TestGreaterLowerThan(t *testing.T) {
+
+	result := greaterLowerEqualThan("3", "3", ">=")
+
+	if !result {
+		t.Errorf("greaterLowerThan was incorrect, got false, want true, 3 is greater or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("3", "3", ">")
+
+	if result {
+		t.Errorf("greaterLowerThan was incorrect, got false, want true, 3 is greater or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("4", "3", ">")
+
+	if !result {
+		t.Errorf("greaterLowerThan was incorrect, got false, want true, 4 is greater or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("2", "3", ">")
+
+	if result {
+		t.Errorf("greaterLowerThan was incorrect, got true, want false, 2 is not greater or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("3", "3", "<=")
+
+	if !result {
+		t.Errorf("greaterLowerThan was incorrect, got true, want false, 3 is lower or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("3", "3", "<")
+
+	if result {
+		t.Errorf("greaterLowerThan was incorrect, got true, want false, 3 is lower or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("2", "3", "<")
+
+	if !result {
+		t.Errorf("greaterLowerThan was incorrect, got true, want false, 2 is lower or equal than 3")
+	}
+
+	result = greaterLowerEqualThan("4", "3", "<")
+
+	if result {
+		t.Errorf("greaterLowerThan was incorrect, got true, want false, 4 is not lower or equal than 3")
+	}
+
+}
+
+func TestCompareVersion(t *testing.T) {
+	result := compareVersion("0.7.1", "0.7.1", ">=")
+
+	if !result {
+		t.Errorf("compareVersion was incorrect, got false, want true, 0.7.1 is >= 0.7.1")
+	}
+
+	result = compareVersion("0.7.1", "0.8.1", ">=")
+
+	if !result {
+		t.Errorf("compareVersion was incorrect, got false, want true, 0.8.1 is >= 0.7.1")
+	}
+
+	result = compareVersion("0.7.1", "0.7.8", ">=")
+
+	if !result {
+		t.Errorf("compareVersion was incorrect, got false, want true, 0.7.8 is >= 0.7.1")
+	}
+
+	result = compareVersion("0.7.1", "1.0.0", ">=")
+
+	if !result {
+		t.Errorf("compareVersion was incorrect, got false, want true, 1.0.0 is >= 0.7.1")
+	}
+
+	result = compareVersion("0.7.1", "1.0.0", ">")
+
+	if !result {
+		t.Errorf("compareVersion was incorrect, got false, want true, 1.0.0 is > 0.7.1")
+	}
+
+	result = compareVersion("1.7.1", "0.8.2", ">")
+
+	if result {
+		t.Errorf("compareVersion was incorrect, got true, want false, 0.8.2 is not > 1.7.1")
+	}
+}
+
+func TestValidateVersion(t *testing.T) {
+	result := validateVersion(">= 0.7.1", "0.7.1")
```

Comment:
What if a version is "0.8.0-alpha" ? Or "0.8.0.1" ? Will it crash?

Reply:
Not with 0.8.0.1 but yes with 0.8.0-alpha. Will we use alpha versions?

Reply:
The user can install whatever he wants, no? I'm looking at our versions.go file: iiuc it's looking for pattern "[0-9]\\.[0-9]\\.[0-9]" (note that anything may follow this pattern - I'm not sure if the `FindStringSubmatch` used there truncates the version for the matched expression or takes the whole string like "0.8.0-alpha").
Also, looking at versions.go, version may be "Unknown". In which case I guess it should be considered as not supported.